core/src/validation.jl

```diff
@@ -660,3 +660,39 @@ function valid_subgrid(
 
     return !errors
 end
+
+function valid_demand(
+    node_id::Vector{NodeID},
+    demand::Vector{
+        Vector{LinearInterpolation{Vector{Float64}, Vector{Float64}, true, Float64}},
+    },
+    priorities::Vector{Int},
+)::Bool
+    errors = false
+
+    for (col, id) in zip(demand, node_id)
+        for (demand_p_itp, p_itp) in zip(col, priorities)
+            if any(demand_p_itp.u .< 0.0)
+                @error "Demand of user node $id with priority #$p_itp should be non-negative"
```

Comment:
```suggestion
                @error "Demand of user node $id with priority $p_itp should be non-negative"
```
We only prefix node IDs with a `#`.